lib/Service/OpenAiAPIService.php

```diff
@@ -252,8 +255,21 @@ public function isQuotaExceeded(?string $userId, int $type): bool {
 			$this->logger->warning('Could not retrieve quota usage for user: ' . $userId . ' and quota type: ' . $type . '. Error: ' . $e->getMessage());
 			throw new Exception('Could not retrieve quota usage.', Http::STATUS_INTERNAL_SERVER_ERROR);
 		}
-
-		return $quotaUsage >= $quota;
+		if ($quotaUsage >= $quota) {
+			$cache = $this->cacheFactory->createLocal(Application::APP_ID);
+			if ($cache->get('quota_exceeded_' . $userId . '_' . $type) === null) {
+				$notification = $this->notificationManager->createNotification();
+				$notification->setApp(Application::APP_ID)
+					->setUser($userId)
+					->setDateTime(new DateTime())
+					->setObject('quota_exceeded', (string)$type)
+					->setSubject('quota_exceeded', ['type' => $type]);
+				$this->notificationManager->notify($notification);
+				$cache->set('quota_exceeded_' . $userId . '_' . $type, true, 3600);
+			}
+			return true;
+		}
+		return false;
```

Comment:
[optional]
wdyt of converting `if ($quotaUsage >= $quota) {` to a negative check and returning early?
```php
if ($quotaUsage < $quota) {
    return false;
}
```

then the pattern becomes `return false`, `return false`, ..., `return true` which then becomes much easier to read and understand.